### PR TITLE
Fixes for #236 and NullPointerException possibility

### PR DESCRIPTION
**Pull request by Xechon**
Haven't mastered github yet, let me know if I did anything wrong here.

### CODE REVIEW

File: src/main/java/com/teamwizardry/wizardry/api/spell/SpellData.java [hunk 1]
```diff
@@ -81,19 +81,20 @@ public void addAllData(HashMap<DataField<?>, Object> data) {
...
 	}
 
 	@Nonnull
-	@SuppressWarnings("unchecked")
-	public <T> T getData(@Nonnull DataField<T> key, @Nonnull T def) {
-		Object value = data.get(key);
-		if (key.getDataType().isInstance(value))
-			return (T)value;
-		return def;
+	public <T> T getDataWithFallback(@Nonnull DataField<T> key, @Nonnull T fallback) {
```
Comment (Avatair) [inline src/main/java/com/teamwizardry/wizardry/api/spell/SpellData.java]: Nonnull should be removed for fallback parameter, too.

**Comment by Xechon** [comment]
> @Avatair commented on this pull request.
> In src/main/java/com/teamwizardry/wizardry/api/spell/SpellData.java:
>  	@Nonnull
> -	@SuppressWarnings("unchecked")
> -	public <T> T getData(@Nonnull DataField<T> key, @Nonnull T def) {
> -		Object value = data.get(key);
> -		if (key.getDataType().isInstance(value))
> -			return (T)value;
> -		return def;
> +	public <T> T getDataWithFallback(@Nonnull DataField<T> key, @Nonnull T fallback) {
> Nonnull should be removed for fallback parameter, too.

Sorry, I couldn't find the actual comment on github, but I copied it as a quote from the email notification I was sent. 
Here is my case for the way I did things: If the fallback parameter is nullable, then we'll have to wrap values for primitives and perform null checking on results, and if you want it to return a null value on a lookup failure, you can just call the non-fallback version directly, instead of telling it to fallback to null. I hope that all made sense.

**Comment by TideSinger** [comment]
Is this PR still active?

**Comment by Xechon** [comment]
> Is this PR still active?

I haven't touched it in a while, due to the lack of feedback or merging. No clue if they have already incorporated fixes for this stuff in the meanwhile.

**Comment by SwissCheese5** [comment]
Our bad. We've been very very busy.

**Comment by SwissCheese5** [comment]
Can you please revert your first two commits? They've been resolved on our end. Your fixes however for the data checking and the edge block freeze most likely work.

**Comment by Xechon** [comment]
Should be good to go, thanks for getting back to me.

**Comment by SwissCheese5** [comment]
@Xechon can you please fix the merge conflict? Once that's fixed, this will be merged.

**Comment by Xechon** [comment]
Thanks for being patient with me. The conflict was that you had fixed the same issue yesterday, but I went ahead and kept my implementation because it avoids code duplication and required the fewest changes to the PR.

**Comment by SwissCheese5** [comment]
uh, I think you re-reverted ur reverted commits

**Pull request merged**